src/Plugin/Visualisation/Style/AxisChart.php

```diff
@@ -738,6 +738,11 @@ protected function chartBuildSettings(array $records) {
       $settings['chart']['data']['columns'][] = array_merge([$field], $this->getSourceFieldValues($field));
     }
 
+    // Override fields labels if set in chart options.
+    if (!empty($this->configuration['data']['field_labels'])) {
```

Comment:
The data names should always be set regardless. The helper method `fieldLabel` will take care of overriding the labels if required, but it will always return all the labels.

```
// Override fields labels if set in chart options.
$settings['chart']['data']['names'] = $this->fieldLabels();
```

Also, everything is so encapsulated at this point that you shouldn't need to know that `fieldLabel` uses `$this->configuration['data']['field_labels']`.

Reply:
So @jorgegc there is no need to do a `!empty(...)` check, as there will always be `field_labels` returned / set?

Reply:
@rob-duplock there is no need because `VisualisationBaseStyle::fieldLabels` works as an API and takes care of what needs to be checked and done in order to return the field labels.